src/openrct2/object/RideObject.cpp

```diff
@@ -229,16 +231,32 @@ void RideObject::Load()
             uint32_t imageIndex = baseImageId;
             carEntry->base_image_id = baseImageId;
 
-            for (uint8_t spriteGroup = 0; spriteGroup < EnumValue(SpriteGroupType::Count); spriteGroup++)
+            switch (carEntry->PaintStyle)
             {
-                if (carEntry->SpriteGroups[spriteGroup].Enabled())
-                {
-                    carEntry->SpriteGroups[spriteGroup].imageId = imageIndex;
-                    const auto spriteCount = carEntry->base_num_frames
-                        * carEntry->NumRotationSprites(static_cast<SpriteGroupType>(spriteGroup))
-                        * SpriteGroupMultiplier[spriteGroup];
-                    imageIndex += spriteCount;
-                }
+                case VEHICLE_VISUAL_RIVER_RAPIDS:
+                case VEHICLE_VISUAL_VIRGINIA_REEL:
+                    // Paint code for these rides do not use sprite groups but #17909 requires them. Dummy sprite groups are
+                    // added.
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::SlopeFlat)] = { baseImageId, SpritePrecision::Sprites16 };
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::Slopes12)] = {
+                        baseImageId + RiverRapidsNumSpinningSprites, SpritePrecision::Sprites4
+                    };
+                    carEntry->SpriteGroups[EnumValue(SpriteGroupType::Slopes25)] = {
+                        baseImageId + RiverRapidsNumSpinningSprites * (1 + NumOrthogonalDirections), SpritePrecision::Sprites4
```

Comment:
Are you sure this is correct? I’m especially talking about the `(1 + NumOrthogonalDirections)` bit.

**If** it is correct, I’d like to see it written as this:
```suggestion
                        baseImageId + (RiverRapidsNumSpinningSprites * (NumOrthogonalDirections + 1)), SpritePrecision::Sprites4
```

Reply:
It's supposed to be 40 so it should be correct.